jetstream/core/orchestrator.py

```diff
@@ -656,6 +680,24 @@ def _generate_thread(self, idx: int):
           block |= not self._transfer_backlogs[idx].empty()
         try:
           new_request = my_generate_backlog.get(block=block, timeout=1.0)
+          if new_request is None:
+            break
+          new_request.metadata.generate_dequeue_time = time.perf_counter()
+          if (
+              self._metrics_collector
+              and new_request.metadata.start_time is not None
+          ):
+            self._metrics_collector.get_queue_duration().observe(
+                # Time in prefill queue
```

Comment:
This could break down to 2 parts: 1: Time for grpc client to server (before prefill enqueue, mostly network latency); 2: Time in prefill queue = new_request.metadata.prefill_dequeue_time - new_request.metadata.prefill_enqueue_time

Reply:
I think you're describing the wait time metric. This metric is only to record time spent enqueued. Sent you the metrics proposal doc with metrics definitions